drivers/hmis/app/graphql/types/hmis_schema/query_type.rb

```diff
@@ -236,6 +236,31 @@ def denied_pending_referral_postings(**args)
       scoped_referral_postings(postings, **args)
     end
 
+    field :merge_candidates, Types::HmisSchema::ClientMergeCandidate.page_type, null: false
+    def merge_candidates
+      raise 'not allowed' unless current_user.can_merge_clients?
+
+      destination_id_to_source_ids = GrdaWarehouse::WarehouseClient.
+        where(data_source_id: current_user.hmis_data_source_id).
+        joins(:source). # drop non existent source clients
+        group(:destination_id).
+        having('count(*) > 1').
+        select('"destination_id", array_agg("source_id") as source_ids').
+        map { |r| [r.destination_id, r.source_ids] }.
+        to_h
+
+      source_lookup = Hmis::Hud::Client.
+        where(id: destination_id_to_source_ids.values.flatten).
+        index_by(&:id)
```

Comment:
I guess I'd be inclined to batch as described above if we do something like this, or at least loop over batches of `destination_id_to_source_ids` so we prevent instantiating thousands of full client records.